src/ui/public/index_patterns/unsupported_time_patterns.js

```diff
@@ -0,0 +1,34 @@
+import { BoundToConfigObjProvider } from 'ui/bound_to_config_obj';
+
+export function UnsupportedTimePatternsProvider(Private, $injector) {
+  const BoundToConfigObj = Private(BoundToConfigObjProvider);
+  const sessionStorage = $injector.get('sessionStorage');
+  const kbnUrl = $injector.get('kbnUrl');
+  const Notifier = $injector.get('Notifier');
+
+  const notify = new Notifier({ location: 'Index Patterns' });
+
+  const HISTORY_STORAGE_KEY = 'indexPatterns:warnAboutUnsupportedTimePatterns:history';
+  const FLAGS = new BoundToConfigObj({
+    enabled: '=indexPatterns:warnAboutUnsupportedTimePatterns'
+  });
+
+  return function warnAboutUnsupportedTimePattern(indexPattern) {
+    if (!FLAGS.enabled) {
+      return;
+    }
+
+    const previousIds = sessionStorage.get(HISTORY_STORAGE_KEY) || [];
+    if (previousIds.includes(indexPattern.id)) {
+      return;
+    }
+
+    sessionStorage.set(HISTORY_STORAGE_KEY, [...previousIds, indexPattern.id]);
+    notify.warning(
```

Comment:
I don't think we should tightly couple application logic (whether things should be displayed) with view logic (how things are displayed).  In the vein, I think it makes sense to uncouple this module from `notify` and `kbnUrl`. Then the caller would have the option of surfacing the message via notifications, inside of an InfoPanel, or in some other part of the UI, depending on what makes the most sense in the given context.

From there, we probably don't need to dictate the precise notification message. The caller should be responsible for determining how to communicate information to the user, again based on the context. At that point, we're left with what I see as the main concern of this module, which is determining whether the message should be displayed at all, based on our flag and persistence rules. This will simplify the code and the unit tests significantly.

So I'd rename this module to be `is_unsupported_time_pattern` and have it return true or false. What do you think?